Server/mods/deathmatch/logic/CPlayerManager.h

```diff
@@ -40,6 +41,7 @@ class CPlayerManager
 
     CPlayer* Get(const NetServerPlayerID& PlayerSocket);
     CPlayer* Get(const char* szNick, bool bCaseSensitive = false);
+    const CPlayer* GetBySerial(const std::string serial) const noexcept;
```

Comment:
`const CPlayer*` is unnecessary. Use `CPlayer*`

Reply:
An overload would be better here instead of removing constness.

Reply:
`GetBySerial` is ok here. An overload is evil here.

Reply:
Why is it evil in your opinion?

Reply:
Because `Get` searches by nickname for a string argument currently. Two different behavior for different string types are evil

Reply:
> huh? What are you talking about?

About your purpose to use overload.

> It should return const CPlayer* in const context and normal CPlayer* in non-const context. This is not evil. In fact, its standard practice that you dont seem to use resulting in making everything non-const because everything is non-const. **That** is evil

This method returns `const CPlayer*` for any context. It request a const_cast or an additional method to modify returned value here. This method will work in const context too. `const` is redundant here.

Reply:
> This method returns `const CPlayer*` for any context. It request a const_cast or an additional method to modify returned value here. This method will work in const context too. `const` is redundant here.

You dont use const_cast because you are not supposed to modify the object. **Thats the point!**
If you want to modify the object, you dont use const. **Thats the point!**
Its good practice to mark it as const if you dont intent to modify whatever it returns (for pointers that is)

Reply:
There is no reason why `GetBySerial` should return a non-modifiable object. `GetBySerial` is const itself and it's enough.

> If you want to modify the object, you dont use const. **Thats the point!**

So you purpose an additional method `GetBySerialNonConst` to return a modifiable object.

Reply:
I'll just go with the solution suggested by TheNormalnij.
I don't think we should put so much emphasis on it, it's really not that interesting to debate..